packages/next/server/next-server.ts

```diff
@@ -732,7 +732,8 @@ export default class NextNodeServer extends BaseServer {
                   __flight__: query.__flight__,
                 } as NextParsedUrlQuery)
               : query),
-            ...(params || {}),
+            // For appDir params is excluded.
+            ...((this.nextConfig.experimental.appDir ? {} : params) || {}),
```

Comment:
Should this check be request specific (app request vs page request) in case `pages` is present as well?

Reply:
Oh yeah it should be for `app` only 😅 Will update

Reply:
Added!